### PR TITLE
Fix message storm when using multiple peers with multicast

### PR DESCRIPTION
**Pull request by OlivierHecart**
The first commit avoids the declare message loop that occurred when using multiple peers with multicast.

Then this PR prevents any routing from/to multicast groups from the same process (router) as this can easily lead to messages loop.

### CODE REVIEW

File: zenoh/src/net/routing/hat/client/mod.rs [hunk 1]
```diff
@@ -275,10 +275,8 @@ impl HatBaseTrait for HatCode {
         _expr: &mut RoutingExpr,
     ) -> bool {
         src_face.id != out_face.id
-            && match (src_face.mcast_group.as_ref(), out_face.mcast_group.as_ref()) {
-                (Some(l), Some(r)) => l != r,
-                _ => true,
-            }
+            && out_face.mcast_group.is_none()
```
Comment (wyfo) [inline zenoh/src/net/routing/hat/client/mod.rs:278]: why this change?

**Review by sashacmc — APPROVED** [review]

File: zenoh/src/net/routing/hat/client/mod.rs (hunk 1 above)
Comment (Mallets) [reply zenoh/src/net/routing/hat/client/mod.rs:278]: It's to avoid sending data to a multicast group when in client mode.
This is also to align with zenoh-pico where joining a multicast group can only be done when in peer mode.

**Review by Mallets — APPROVED** [review]
